cmake/external/helper_functions.cmake

```diff
@@ -0,0 +1,193 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
+
+# This file was copied from cmake source with modifications
```

Comment:
> This file was copied from cmake source with modifications [](http://example.com/codeflow?start=1&length=58)

What's the purpose of the copy?

Reply:
1. Add the EXCLUDE_FROM_ALL keyword when this function calls add_subdirectory. It will also resolve the 'make install' issue.
2. Group the VC projects into the "external" folder. We can do it at there in a centralized way instead of doing it one by one.
3.  Set the cmake property `COMPILE_WARNING_AS_ERROR` to OFF for these external projects.